adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,34 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source){
+	std::stack<char> stack;
+	auto find_pair = [](char c){
+		for (auto it = brackets.cbegin() ; it != brackets.cend(); it++) {
```

Comment:
Always prefer range for loops, not sure that you should operate with iterators here, try to use a bool named condition for that